compiler/test/dotty/tools/dotc/CompilationTests.scala

```diff
@@ -284,6 +284,13 @@ class CompilationTests extends ParallelTesting {
     implicit val testGroup: TestGroup = TestGroup("explicitNullsRun")
     compileFilesInDir("tests/explicit-nulls/run", explicitNullsOptions)
   }.checkRuns()
+
+  @Test def compilerCrash: Unit = {
```

Comment:
This is test duplication, if you put the test file in `tests/pos/7965.scala` then it will be tested exactly the same by `pos` in `CompilationTests`, you can then run it exclusively with `sbt testCompilation tests/pos/7965.scala`. http://dotty.epfl.ch/docs/contributing/testing.html

This test case can only be accepted in a PR if it actually passes the test. If the intention is to include code in the repository to prove the failure to compile then the example being documented in the issue is enough for someone to begin a fix.